test/images/agnhost/netexec/netexec.go

```diff
@@ -125,7 +144,14 @@ func startHTTPServer(httpPort int) {
 	// older handlers
 	http.HandleFunc("/hostName", hostNameHandler)
 	http.HandleFunc("/shutdown", shutdownHandler)
-	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", httpPort), nil))
+
+	if err := server.ListenAndServe(); err != nil {
+		if err == http.ErrServerClosed {
+			// wait until the goroutine calls os.Exit()
+			select {}
```

Comment:
maybe do this to protect against hanging forever if exitCh wasn't what triggered the shutdown?
```
select {
case <-exitCh:
  // graceful exit, wait until the goroutine calls os.Exit()
  select {}
default:
  // exiting for some other reason
}
```

Reply:
oh, exitCh isn't a close channel... hmm